code/game/turfs/flooring/_flooring_decals.dm

```diff
@@ -13,12 +13,18 @@ var/global/list/floor_decals = list()
 	var/detail_overlay
 	var/detail_color
 
-// Have to wait for turfs to set up their flooring, so we can better guess at our layers.
-/obj/effect/floor_decal/Initialize()
-	..()
-	return INITIALIZE_HINT_LATELOAD
+/obj/effect/floor_decal/Initialize(mapload)
+	SHOULD_CALL_PARENT(FALSE)
+	SHOULD_NOT_OVERRIDE(TRUE)
+	return INITIALIZE_HINT_NORMAL
+
+/obj/effect/floor_decal/New(loc, var/newdir, var/newcolour, var/newappearance)
```

Comment:
I don't like this being `New()`, there shouldn't be enough of these for the overhead saved here to be remotely meaningful.

I barely justify merging atom/New() into turf/New() on aurora.

Reply:
for post-roundstart maploading like we do downstream it's actually pretty significant, non-mapload initatoms is pretty damn slow and ships tend to have a bunch of decals. normalized so that both traces have the same number of calls, it saved 87.25%
![image](https://github.com/user-attachments/assets/455197b8-ba19-4297-911f-dc6cb6a054c9)
also consider lush exoplanets; `/datum/random_map/noise/exoplanet/proc/spawn_grass` spawns a *lot* of floor decals. i tested 3 lush exoplanets and compared with just an optimized `/Initialize` compared to `/New`, and with `/New`, `spawn_grass` is 71.92% cheaper.
![image](https://github.com/user-attachments/assets/d02d2904-7129-4723-acc5-56c913bb1b30)

Reply:
1 second of init time is nothing, I don't really care about the percentage improvement if the absolute init time is already very fast.

Reply:
it's kind of hard to profile this because I can't exactly profile the missing InitAtom calls due to this change, but i think it's pretty significant. i'll profile against a larger map like Tether at some point

Reply:
@Lohikar i believe this particular change was an 8 second (out of ~64 seconds) time save on Tether, which is pretty significant imo. also with this PR as a whole Shaded Hills went from around 42 seconds to 19 seconds in local testing, though idk if this particular change is as relevant there